Sources/GRPCReflectionService/Server/ReflectionService.swift

```diff
@@ -112,8 +119,28 @@ internal struct ReflectionServiceData: Sendable {
               """
           )
         }
+        if self.fieldNumbersByType[typeName] == nil {
+          self.fieldNumbersByType[typeName] = []
+        }
+        let numberPosition = self.fieldNumbersByType[typeName]!.count
+        self.fieldNumbersByType[typeName]?.insert(
+          `extension`.number,
+          at: numberPosition
+        )
```

Comment:
A couple of notes:
1. Don't use `insert` if you want to add an element to the end of the array, use `append`
2. We don't need to check whether there's a value in the dictionary and then insert an empty array. Instead we can use https://developer.apple.com/documentation/swift/dictionary/subscript(_:default:)